### DISCUSSION TITLE
Create cluster error

### DISCUSSION THREAD

**Discussion by LiJie2324** [microsoft/garnet#706] [Q&A]
Hello, i have a problem, if add  key-value {    "AuthenticationMode":"Password",	"Password":"123456" }    in the   garnet.conf ,
then command create cluster will error,  **the command output** :
![bug ](https://github.com/user-attachments/assets/4219e83a-6e2e-4c93-b2b0-9fb981c51e8c)

**the garnet.conf ：**
![image](https://github.com/user-attachments/assets/8a61c8bd-a516-498a-b476-29121b467ca1)

**Comment by vazois** [comment]
You have to provide --cluster-username and --cluster-password for authentication between nodes to work

**Reply by LiJie2324** [reply]
hello , thank you for your reply。
I try to provide  --cluster-username and --cluster-password , but it also can't to work 
![image](https://github.com/user-attachments/assets/a390978e-e73d-4a83-8ac1-7d2d064b364f)
![image](https://github.com/user-attachments/assets/4e45f6a2-06bb-42b9-a03d-7181c41e1c53)
![image](https://github.com/user-attachments/assets/68e6dc28-a190-47a4-a78e-ec757b6dcb0a)

**Comment by vazois** [comment]
I am not able to reproduce this. Could you try to provide the parameters in command line when you are starting up the instances?
For example:
GarnetServer.exe --port=7000 --bind= --logger-level=Trace --cluster=True --password=1234 --auth=Password --cluster-password=1234

**Comment by LiLuke2324** [comment]
Hello, if I retry providing the parameters in the command and the --password does not match --cluster-password, using redis-cli.exe to create the cluster will result in an error. However, if --password matches --cluster-password, the cluster creation will succeed.
redis-cli.exe command:
`./redis-cli.exe -a 1234 --cluster create 127.0.0.1:6379 127.0.0.1:6380 127.0.0.1:6381 127.0.0.1:7000 --cluster-yes`

**error log  message :**
03::36::30 fail: ClusterManager[0] AUTH returned error System.Exception: WRONGPASS Invalid password    at Garnet.client.GarnetClient.ConnectAsync(CancellationToken token)
03::36::30 fail: ClusterManager[0] Meet terminated with error System.Exception: WRONGPASS Invalid password    at Garnet.client.GarnetClient.ConnectAsync(CancellationToken token)    at Garnet.client.GarnetClient.ReconnectAsync(CancellationToken token)    at Garnet.cluster.GarnetServerNode.Initialize()    at Garnet.cluster.ClusterManager.Meet(String address, Int32 port)
03::38::11 dbug: Session[0] [127.0.0.1:56523] [034E5D84] Disposing RespServerSession Id=1
![image](https://github.com/user-attachments/assets/9ca55fe9-2c46-439c-b588-20dcdef15927)

**Reply by vazois** [reply]
I did not understand the last response. Did you try providing the server config in command line and succeeded? I am trying to figure out if there is an issue with the config file or not. Thanks!

**Reply by LiJie2324** [reply]
hello, if I use [config1]  ,redis-cli create cluster  will success, if  use [config2],  ,redis-cli create cluster  will error.
**redis-cli command example:**
`./redis-cli.exe -a 1234 --cluster create 127.0.0.1:6379 127.0.0.1:6380 127.0.0.1:6381 127.0.0.1:7000 --cluster-yes`
config example:
**config1:**
`{
	"Address":"127.0.0.1",
	"Port" : 6379,
	"LogLevel":"Debug",
	"EnableCluster":true,
	"AuthenticationMode":"Password",
	"Password": "1234",
	"ClusterPassword":"123456"
}`
**Config2:**
`{
	"Address":"127.0.0.1",
	"Port" : 6379,
	"LogLevel":"Debug",
	"EnableCluster":true,
	"AuthenticationMode":"Password",
	"Password": "1234",
	"ClusterPassword":"1234"
}`

**Reply by vazois** [reply]
This is expected behavior, the password provided for the cluster node to connect to another node, should match the AUTH credentials of the remote node. FYI, using ACL AuthenticationMode is preferred way to manage passwords but you will still have to provide the ClusterPassword option for the node to be aware how to connect with the remote cluster nodes
